apps/src/signUpFlow/FinishTeacherAccount.tsx

```diff
@@ -0,0 +1,109 @@
+
+import Checkbox from '../componentLibrary/checkbox/Checkbox';
+
+import style from './finish-teacher-account.module.scss';
+
+const USER_NAME_SESSION_KEY = 'username';
+const EMAIL_OPT_IN_SESSION_KEY = 'emailOptIn';
+
+const FinishTeacherAccount: React.FunctionComponent<{
+  usIp: boolean;
+}> = ({usIp}) => {
+  const [name, setName] = useState('');
+  const [emailOptInChecked, setEmailOptInChecked] = useState(false);
+
+  const onNameChange = (e: React.ChangeEvent<HTMLInputElement>): void => {
+    const newName = e.target.value;
+    setName(newName);
+    sessionStorage.setItem(USER_NAME_SESSION_KEY, newName);
+  };
+
+  const onEmailOptInChange = (): void => {
+    const newOptInCheckedChoice = !emailOptInChecked;
+    setEmailOptInChecked(newOptInCheckedChoice);
+    sessionStorage.setItem(
+      EMAIL_OPT_IN_SESSION_KEY,
+      `${newOptInCheckedChoice}`
+    );
+  };
+
+  return (
+    <div className={style.finishTeacherAccountContainer}>
+      <div className={style.headerTextContainer}>
+        <Heading2>{locale.finish_teacher_account_title()}</Heading2>
+        <BodyTwoText>{locale.finish_teacher_account_subtitle()}</BodyTwoText>
+      </div>
+      <div className={style.inputContainer}>
+        <BodyTwoText
+          className={style.labelText}
+          visualAppearance={'heading-xs'}
+        >
+          {locale.account_name_input()}
+        </BodyTwoText>
+        <input
+          type="text"
+          name="[user][name]"
+          onChange={onNameChange}
+          value={name}
+        />
+        <BodyThreeText>
+          {locale.finish_teacher_account_name_input_desc()}
+        </BodyThreeText>
+        <SchoolDataInputs usIp={usIp} includeHeaders={false} />
+        <div className={style.emailOptInContainer}>
+          <BodyTwoText
+            className={style.labelText}
+            visualAppearance={'heading-xs'}
+          >
+            {locale.stay_updated_email_opt_in()}
+          </BodyTwoText>
+          <span className={style.emailOptInCheckboxContainer}>
+            <div className={style.emailOptInCheckbox}>
+              <Checkbox
+                label=""
+                checked={emailOptInChecked}
+                onChange={onEmailOptInChange}
+                name="[user][emailOptIn]"
+              />
+            </div>
+            <BodyTwoText
+              className={style.emailOptInLabel}
+              visualAppearance={'body-two'}
+            >
+              {locale.stay_updated_email_opt_in_desc()}
+            </BodyTwoText>
+          </span>
+          <BodyThreeText className={style.emailOptInFootnote}>
+            {locale.stay_updated_email_opt_in_footnote()}
+          </BodyThreeText>
+        </div>
+      </div>
+      <div className={style.finishSignUpButtonContainer}>
+        <Button
+          className={style.finishSignUpButton}
+          color={buttonColors.purple}
+          type="primary"
+          onClick={() => console.log('FINISH SIGN UP')}
```

Comment:
Currently just a placeholder `log` statement for the button. This will be filled in when the other sign-up pages are all put together!